### ISSUE TITLE
[FEATURE] Add a way to access logs

### ISSUE THREAD

**Issue by Zabuzard** [Together-Java/TJ-Bot#103]
We need a way to be able to read the logs of the two bots, without actually logging in to the VPS and reading the text files.

This could for example be a full business solution like Kibana or maybe also just another routine in the bot that sends all log messages to a dedicated Discord channel.

Also, it would be good to have a way to change the logging level hot (while the application is running) - or at least without logging in to the VPS and changing the config manually.

**Comment by d3vel0per** [comment]
I see 3 options in my mind:
* Spring's ecosystem brings in support of the mentioned functionality and a lot more (one of the reasons I was more inclined to D4J). So we spend some time in integrating JDA with spring
* Use something like https://github.com/portainer/portainer which is more of a docker management UI but achieves our current goal and a few others like bringing down a bot if needed.
* Roll out our own implementation using a combination of Elastic-Logstash-Kibana.

Others can chip in

**Comment by Tais993** [comment]
Or abuse the logback appender to send it to a channel :p

**Comment by marko-radosavljevic** [comment]
@d3vel0per Any updates on this one?

**Comment by Zabuzard** [comment]
I like channel logging and hot log level change per discord command.

Unfortunately, that does not help if the bot is failung during JDA setup (like it is now, after the Java 17 migration).

**Comment by krankkkk** [comment]
The most simplistic implementation would be to store the logevents in the DB and query them on demand per  some command. That way you don't hit rate-Limits because you are constantly pushing to some Discord-Log-Channel and you also don't rely on full Business solutions.

**Comment by d3vel0per** [comment]
> @d3vel0per Any updates on this one?

Not really, I was waiting for more ideas. I was evaluating a few cloud products but the group didn't have a consensus on those.
Streaming back to a channel sounds interesting with the mentioned already mentioned caveats (may be its not a part of the bot itself)
what @krankkkk suggested also sounds good but then we will have to build a lot of infra around it viz. archiving, well-structured logs, etc.
or we may host a full log analysis stack (but i think the setup itself will become more costly than the bot itself)

**Comment by d3vel0per** [comment]
> @d3vel0per Any updates on this one?

Not really, I was waiting for more ideas. I was evaluating a few cloud products but the group didn't have a consensus on those.
Streaming back to a channel sounds interesting with the mentioned already mentioned caveats (may be its not a part of the bot itself)
what @krankkkk suggested also sounds good but then we will have to build a lot of infra around it viz. archiving, well-structured logs, etc.
or we may host a full logging stack (but i think the setup itself will become more costly than the bot itself)

**Comment by marko-radosavljevic** [comment]
For now, anything would work, currently only 2 people have access to logs, which is not optimal.

Can I get your opinion on this @I-Al-Istannen? 😇

**Comment by d3vel0per** [comment]
Haven't tried before but https://github.com/mthenw/frontail seems to be a lightweight alternative than other full fledged solutions

**Comment by I-Al-Istannen** [comment]
We don't really need any historical logging or archiving and so on and are probably fine with just text based logs as a start IMHO. Anything that can stream a text file to users sounds good enough and discord oauth integration would be awesome.

I am not sure if there is any easy program we can use currently.
